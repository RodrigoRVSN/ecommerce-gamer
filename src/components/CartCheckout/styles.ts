import styled from "styled-components";

export const CartContainer = styled("div")`
  display: flex;
  flex-direction: column;
  width: 95vw;
  height: 100%;
  background: #0077b6;
  color: #a4161a;
  text-align: center;
  font-size: 1.3rem;
`;

export const ButtonContainer = styled("div")`
  display: flex;
  justify-content: center;
  gap: 5vw;
`;
