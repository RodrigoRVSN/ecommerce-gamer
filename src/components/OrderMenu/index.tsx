import { Button, Menu, MenuItem } from "@material-ui/core";

import React from "react";
import { useCart } from "../../hooks/useCart";

export function OrderMenu() {
  const { NameToA, NameToZ, PriceHigher, PriceLower, ScoreHigher, ScoreLower } =
    useCart();
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <>
      <Button
        aria-controls="simple-menu"
        aria-haspopup="true"
        onClick={handleClick}
        variant="contained"
        size="large"
        color="primary"
      >
        <img src="/assets/arrow-down-icon.svg" alt="filtro" />
      </Button>
      <Menu
        id="simple-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        <MenuItem
          onClick={() => {
            NameToZ();
            handleClose();
          }}
        >
          Ordem Alfabética (A-Z)
        </MenuItem>
        <MenuItem
          onClick={() => {
            NameToA();
            handleClose();
          }}
        >
          Ordem Alfabética (Z-A)
        </MenuItem>
        <MenuItem
          onClick={() => {
            PriceHigher();
            handleClose();
          }}
        >
          Preço (Maior - Menor)
        </MenuItem>
        <MenuItem
          onClick={() => {
            PriceLower();
            handleClose();
          }}
        >
          Preço (Menor - Maior)
        </MenuItem>
        <MenuItem
          onClick={() => {
            ScoreHigher();
            handleClose();
          }}
        >
          Pontuação (Maior - Menor)
        </MenuItem>
        <MenuItem
          onClick={() => {
            ScoreLower();
            handleClose();
          }}
        >
          Pontuação (Menor - Maior)
        </MenuItem>
      </Menu>
    </>
  );
}
