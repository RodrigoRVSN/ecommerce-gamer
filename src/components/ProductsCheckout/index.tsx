import { Button, Card } from "@material-ui/core";

import {
  HeaderItem,
  CheckoutContainer,
  ItemsContainer,
  InfoProduct,
} from "./styles";

import { useCart } from "../../hooks/useCart";
import DeleteSweepIcon from "@material-ui/icons/DeleteSweep";
import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
import Badge from "@material-ui/core/Badge";

export function ProductsCheckout() {
  const { addItem, removeItem, itemSelected } = useCart();

  const uniqueItens = itemSelected?.filter(
    (el, i, arr) => arr.indexOf(el) === i
  );

  return (
    <>
      <ItemsContainer>
        {uniqueItens?.map((item, index) => {
          return (
            <Card>
              <CheckoutContainer>
                <InfoProduct>
                  <img src={`/assets/${item.image}`} alt={item.name} />
                  <h4>{item.name}</h4>
                </InfoProduct>
                <HeaderItem>
                  <h3>{`R$ ${item.price.toLocaleString("PT")}`}</h3>
                  <Badge badgeContent={item.amount} color="primary">
                    <ShoppingCartIcon />
                  </Badge>
                  <Button
                    disabled={item.amount < 1}
                    variant="contained"
                    onClick={() => removeItem(item)}
                    color="primary"
                    size="small"
                  >
                    <DeleteSweepIcon />
                  </Button>
                  <Button
                    variant="contained"
                    onClick={() => addItem(item)}
                    color="secondary"
                  >
                    <img
                      width="25rem"
                      height="25rem"
                      src="/assets/cart-icon.svg"
                      alt="carrinho"
                    />
                  </Button>
                </HeaderItem>
              </CheckoutContainer>
            </Card>
          );
        })}
      </ItemsContainer>
    </>
  );
}
