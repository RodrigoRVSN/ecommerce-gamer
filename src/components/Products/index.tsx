import { Button, Card } from "@material-ui/core";

import { ProductContainer, Title, CartProduct, HeaderItem } from "./styles";

import Grid from "@material-ui/core/Grid";
import { useCart } from "../../hooks/useCart";
import DeleteSweepIcon from "@material-ui/icons/DeleteSweep";
import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
import Badge from "@material-ui/core/Badge";

export function Products() {
  const { addItem, removeItem, allItems } = useCart();

  return (
    <>
      {allItems?.map((item, index) => {
        return (
          <Grid key={index} item xs={11} md={4} lg={3}>
            <Card>
              <HeaderItem>
                <h3># {item.id}</h3>
                <Badge badgeContent={item.amount} color="primary">
                  <ShoppingCartIcon />
                </Badge>
              </HeaderItem>
              <ProductContainer>
                <img src={`/assets/${item.image}`} alt={item.name} />
                <Title>{item.name}</Title>
                <h3>Pontuação: {item.score}</h3>
                <CartProduct>
                  <h3>{`R$ ${item.price.toLocaleString("PT")}`}</h3>
                  <Button
                    disabled={item.amount < 1}
                    variant="contained"
                    onClick={() => removeItem(item)}
                    color="primary"
                  >
                    <DeleteSweepIcon />
                  </Button>
                  <Button
                    variant="contained"
                    onClick={() => addItem(item)}
                    color="secondary"
                  >
                    <img
                      width="25rem"
                      height="25rem"
                      src="/assets/cart-icon.svg"
                      alt="carrinho"
                    />
                  </Button>
                </CartProduct>
              </ProductContainer>
            </Card>
          </Grid>
        );
      })}
    </>
  );
}
