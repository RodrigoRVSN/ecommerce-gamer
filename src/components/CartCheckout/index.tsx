import { Button, Drawer } from "@material-ui/core";
import { useEffect, useState } from "react";
import { toast } from "react-toastify";
import { useCart } from "../../hooks/useCart";

import { CartContainer, ButtonContainer } from "./styles";
import { ProductsCheckout } from "../../components/ProductsCheckout";

export function CartCheckout(): JSX.Element {
  const { cartOpen, setCartOpen, price, shipping } = useCart();

  const [totalPrice, setTotalPrice] = useState(0);

  useEffect(() => {
    setTotalPrice(Number(price) + Number(shipping));
  }, [price, shipping, totalPrice]);

  return (
    <>
      <Button onClick={() => setCartOpen(true)}>
        <img src="/assets/cart-icon.svg" alt="carrinho" />
      </Button>
      <Drawer anchor="right" open={cartOpen} onClose={() => setCartOpen(false)}>
        <CartContainer>
          <h2>
            {price > 0
              ? `Valor dos produtos: R$ ${price.toLocaleString("pt")}`
              : "Não há produtos no carrinho 😞"}
          </h2>
          <h2>
            {shipping > 0 ? `Frete: R$ ${shipping},00` : "Não há fretes 😎"}
          </h2>
          <h2>
            {totalPrice > 0 &&
              `Valor total: R$ ${totalPrice.toLocaleString("pt")}`}
          </h2>

          <ProductsCheckout />

          <ButtonContainer>
            <Button
              color="primary"
              variant="contained"
              onClick={() => setCartOpen(false)}
            >
              Fechar
            </Button>
            <Button
              color="secondary"
              variant="contained"
              onClick={() => {
                setCartOpen(false);
                toast.success("✔️ Compra finalizada!");
              }}
            >
              COMPRAR
            </Button>
          </ButtonContainer>
        </CartContainer>
      </Drawer>
    </>
  );
}
