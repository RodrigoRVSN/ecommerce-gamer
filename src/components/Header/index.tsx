import { HeaderContainer, HeaderContent } from "./style";
import { Container } from "@material-ui/core";
import { CartCheckout } from "../CartCheckout";

export function Header(): JSX.Element {
  return (
    <>
      <HeaderContainer>
        <Container>
          <HeaderContent>
            E-GAMER
            <CartCheckout />
          </HeaderContent>
        </Container>
      </HeaderContainer>
    </>
  );
}
