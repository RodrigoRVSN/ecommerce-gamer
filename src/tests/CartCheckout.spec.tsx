import React from "react";
import { render } from "@testing-library/react";
import { CartCheckout } from "../components/CartCheckout";

describe("<CartCheckout />", () => {
  it("should render the checkout drawer", () => {
    const { container } = render(<CartCheckout />);
    expect(container).toMatchSnapshot();
  });
});
