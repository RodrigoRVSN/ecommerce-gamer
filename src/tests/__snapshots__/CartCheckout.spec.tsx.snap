// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCheckout /> should render the checkout drawer 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <img
        alt="carrinho"
        src="/assets/cart-icon.svg"
      />
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
